GoogleMapsApi/Entities/Geocoding/Request/GeocodingComponents.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Generic;
+
+namespace GoogleMapsApi.Entities.Geocoding.Request
+{
+    public class GeocodingComponents : Dictionary<string, string>
```

Comment:
Extends Dictionary? 
IMHO it's better to do composition, i.e. have a Dictionary inside.
When you extend Dictionary you expose all Dictionary's API. For example user can do strange stuff like comp.clear(), or because it's IEnumerable user can enumerate on the key/value pairs ....